java-frontend/src/main/java/org/sonar/java/JavaFrontend.java

```diff
@@ -0,0 +1,643 @@
+        case IDENTIFIER:
+          executeIdentifier((IdentifierTree) tree);
+          break;
+        case MEMBER_SELECT:
+          executeMemberSelect((MemberSelectExpressionTree) tree);
+          break;
+        case INT_LITERAL:
+        case LONG_LITERAL:
+        case FLOAT_LITERAL:
+        case DOUBLE_LITERAL:
+        case CHAR_LITERAL:
+        case STRING_LITERAL:
+        case BOOLEAN_LITERAL:
+        case NULL_LITERAL:
+          executeLiteral();
+          break;
+        case LAMBDA_EXPRESSION:
+        case METHOD_REFERENCE:
+          executeMethodReference();
+          break;
+        case ASSERT_STATEMENT:
+          executeAssertStatement();
+          break;
+        default:
+          throw new IllegalArgumentException("Unhandled tree: " + tree.getClass().getSimpleName() + " - " + tree);
+      }
+    }
+
+    private void executeLiteral() {
+      stack.push(new TaintFreeSource());
+    }
+
+    private void executeIdentifier(IdentifierTree tree) {
+      if (TaintProgramState.isString(tree.symbolType())) {
+        stack.push(state.taintSourceFor(tree.symbol()));
+      } else {
+        stack.push(new TaintFreeSource());
+      }
+    }
+
+    private void executeAssignment(AssignmentExpressionTree tree) {
+      TaintSource ts;
+      if (tree.is(Tree.Kind.ASSIGNMENT)) {
+        ts = stack.pop();
+        if (!ExpressionUtils.isSimpleAssignment(tree)) {
+          stack.pop();
+        }
+      } else {
+        ts = stack.pop();
+        stack.pop();
+      }
+
+      Symbol symbol = null;
+      if (tree.variable().is(Tree.Kind.IDENTIFIER) || ExpressionUtils.isSelectOnThisOrSuper(tree)) {
+        symbol = ExpressionUtils.extractIdentifier(tree).symbol();
+        // TODO How to handle any assignment?
+        state.put(symbol, ts);
+      }
+      stack.push(ts);
+    }
+
+    private void executeMethodInvocation(MethodInvocationTree mit) {
+      List<TaintSource> arguments = new ArrayList<>();
+      for (int i = 0; i < mit.arguments().size(); i++) {
+        arguments.add(stack.pop());
+      }
+      stack.pop();
+
+      // TODO What if void method?
+      stack.push(state.taintSourceFor(mit.symbol(), Lists.reverse(arguments)));
+    }
+
+    private void executeVariable(VariableTree variableTree) {
+      if (variableTree.initializer() != null) {
+        state.put(variableTree.symbol(), stack.pop());
+      } else {
+        state.put(variableTree.symbol(), new TaintFreeSource());
+      }
+    }
+
+    private void executeTypeCast(TypeCastTree typeCast) {
+      // TODO
+      throw new UnsupportedOperationException();
+    }
+
+    private void executeLogicalAssignment(AssignmentExpressionTree tree) {
+      // TODO
+      throw new UnsupportedOperationException();
+    }
+
+    private void executeArrayAccessExpression(ArrayAccessExpressionTree tree) {
+      // TODO
+      throw new UnsupportedOperationException();
+    }
+
+    private void executeNewArray(NewArrayTree newArrayTree) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make "executeNewArray" a "static" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "newArrayTree". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)